cmd/fluent-watcher/fluentbit/main.go

```diff
@@ -213,17 +218,21 @@ func wait() error {
 	mutex.Unlock()
 
 	startTime := time.Now()
+
+	flbTerminated = make(chan bool, 1)
```

Comment:
Why need to create a new channel every time? Like I said before if the `select` goroutine is blocked for some reason, such as CPU high, then the `flbTerminated` may already be a new channel.

The `flbTerminated` will always be consumed by the `select` goroutine, so I think it no need to create a new channel every time.

Reply:
The problem that I see is that in case fluent-bit exits with an error, flbTerminated will then not be consumed and subsequently the process will be blocked until it is consumed. Creating a new channel each time prevents that

Reply:
> The problem that I see is that in case fluent-bit exits with an error, flbTerminated will then not be consumed and subsequently the process will be blocked until it is consumed. Creating a new channel each time prevents that

@wanjunlei Are you ok with @jjsiv‘s explanation

Reply:
> The problem that I see is that in case fluent-bit exits with an error, flbTerminated will then not be consumed and subsequently the process will be blocked until it is consumed. Creating a new channel each time prevents that

Ok, I see what you mean. But how to solve the problem I posed above?

Reply:
I understand that it could be a potential problem under some circumstances (though I guess the likelihood is rather hard to estimate). I've been running it on a few clusters for over a week and it's been flawless so far.
If you believe it's something that should be given more thought I can have a look at it next week and see if I can come up with anything. Or maybe do you/anyone else have any suggestion on how could this be improved? I'm open to any suggestions

Reply:
We only need to ensure that the `wait` function cannot be re-executed before the `stop` function is executed.
We can move the recreation of  `flbTerminated` to the `start` function because the `start` and `stop` functions are mutually exclusive.

https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-watcher/fluentbit/main.go#L184
https://github.com/fluent/fluent-operator/blob/master/cmd/fluent-watcher/fluentbit/main.go#L262

Reply:
Thanks for the suggestion, I did just move it to start().
On a second thought though, wouldn't the mutual exclusion already prevent the situation of wait() creating a new channel before stop() finishes from happening? I did just do a quick test by adding a time.Sleep of 10s to imitate a situation where the execution of the stop() goroutine pauses before select (this is with flbTerminated channel still in wait():
```
level=info time=2023-01-30T22:59:27Z msg="Config file changed, stopping Fluent Bit"
level=info time=2023-01-30T22:59:27Z msg="Sent SIGTERM to FluentBit, waiting max 30s"
level=info time=2023-01-30T22:59:27Z msg="Pausing on stop()..." --- pausing for 10s before select
[2023/01/30 22:59:27] [engine] caught signal (SIGTERM)
[2023/01/30 22:59:27] [ warn] [engine] service will shutdown in max 5 seconds
[2023/01/30 22:59:28] [ info] [engine] service has stopped (0 pending tasks)
level=info time=2023-01-30T22:59:28Z msg=backoff delay=1s
level=info time=2023-01-30T22:59:29Z msg="backoff timer done" actual=1.001111057s expected=1s
level=info time=2023-01-30T22:59:29Z msg="Waiting to start..." --- log before start() mutex, waits for stop() to finish
level=info time=2023-01-30T22:59:37Z msg="FluentBit terminated successfully" --- select in stop() executes after 10s
level=info time=2023-01-30T22:59:37Z msg="Config file changed, stopped Fluent Bit"
level=info time=2023-01-30T22:59:37Z msg="Fluent bit started" --- start() executes after stop unlocks
level=info time=2023-01-30T22:59:37Z msg="New channel" --- new channel created
```

Reply:
Yes, you are right, it is my mistake.